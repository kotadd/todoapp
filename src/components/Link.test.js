import React from 'react';
import { shallow } from 'enzyme';

import Link from './Link';

describe('Link component', () => {
  let wrapper;
  let mockChildren;
  let mockOnClick;

  beforeEach(() => {
    mockChildren = (
      <>
        <span>Children</span>
      </>
    );

    mockOnClick = jest.fn();

    const mockProps = {
      active: false,
      children: mockChildren,
      onClick: mockOnClick
    };

    wrapper = shallow(<Link {...mockProps} />);
  });

  it('should render Link component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when Link is clicked', () => {
    wrapper.find('button').simulate('click');

    expect(mockOnClick).toHaveBeenCalled();
  });
});
