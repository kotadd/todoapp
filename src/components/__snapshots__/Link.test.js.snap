// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component should render Link component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Link
    active={false}
    onClick={[MockFunction]}
  >
    <React.Fragment>
      <span>
        Children
      </span>
    </React.Fragment>
  </Link>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <React.Fragment>
        <span>
          Children
        </span>
      </React.Fragment>,
      "disabled": false,
      "onClick": [MockFunction],
      "style": Object {
        "marginLeft": "4px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <span>
          Children
        </span>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Children",
        },
        "ref": null,
        "rendered": "Children",
        "type": "span",
      },
      "type": Symbol(react.fragment),
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <React.Fragment>
          <span>
            Children
          </span>
        </React.Fragment>,
        "disabled": false,
        "onClick": [MockFunction],
        "style": Object {
          "marginLeft": "4px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span>
            Children
          </span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Children",
          },
          "ref": null,
          "rendered": "Children",
          "type": "span",
        },
        "type": Symbol(react.fragment),
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
